app/controllers/partner_users_controller.rb

```diff
@@ -0,0 +1,93 @@
+
+  def index
+    @users = @partner.profile.users
+    @user = User.new(name: "")
+  end
+
+  def create
+    user = UserInviteService.invite(email: user_params[:email], name: user_params[:name], roles: [Role::PARTNER], resource: @partner.profile)
+    user.valid?
+
+    respond_to do |format|
+      format.turbo_stream do
+        if user.errors.none?
+          flash.now[:notice] = "#{user.name} has been invited. Invitation email sent to #{user.email}"
+
+          render turbo_stream: [
+            turbo_stream.replace("partners/#{@partner.id}/form", partial: "partner_users/form", locals: {partner: @partner, user: User.new(name: "")}),
+            turbo_stream.replace("flash", partial: "shared/flash"),
+            turbo_stream.replace("partners/#{@partner.id}/users", partial: "partner_users/users", locals: {users: @partner.reload.profile.users, partner: @partner})
+          ]
+        else
+          flash.now[:error] = "Invitation failed. Check the form for errors."
+          render turbo_stream: [
+            turbo_stream.replace("partners/#{@partner.id}/form", partial: "partner_users/form", locals: {partner: @partner, user: user}),
+            turbo_stream.replace("flash", partial: "shared/flash")
+          ], status: :bad_request
+        end
+      end
+    end
+  end
+
+  def destroy
+    user = User.find(params[:id])
+
+    respond_to do |format|
+      format.turbo_stream do
+        if user.remove_role(Role::PARTNER, @partner.profile)
+          flash.now[:notice] = "Access to #{user.name} has been revoked."
+          render turbo_stream: [
+            turbo_stream.replace("flash", partial: "shared/flash"),
+            turbo_stream.replace("partners/#{@partner.id}/users", partial: "partner_users/users", locals: {users: @partner.reload.profile.users, partner: @partner})
+          ]
+        else
+          flash.now[:error] = "Invitation failed. Check the form for errors."
+          render turbo_stream: [
+            turbo_stream.replace("flash", partial: "shared/flash")
+          ], status: :bad_request
+        end
+      end
+    end
+  end
+
+  def resend_invitation
+    user = User.find(params[:id])
+
+    if user.invitation_accepted_at.nil?
+      user.invite!
+    else
+      user.errors.add(:base, "User has already accepted invitation.")
+    end
+
+    respond_to do |format|
+      format.turbo_stream do
+        if user.errors.none?
+          flash.now[:notice] = "Invitation email sent to #{user.email}"
+          render turbo_stream: [
+            turbo_stream.replace("flash", partial: "shared/flash"),
+            turbo_stream.replace("partners/#{@partner.id}/users", partial: "partner_users/users", locals: {users: @partner.reload.profile.users, partner: @partner})
+          ]
+        else
+          flash.now[:error] = user.errors.full_messages.to_sentence
+          render turbo_stream: [
+            turbo_stream.replace("flash", partial: "shared/flash")
```

Comment:
This seems like overkill for what's essentially a form. Is there a reason we're jumping straight to turbo streams instead of using Turbo Frames or even Turbo Drive?

Reply:
Hmm I don't know exactly the differences in the options. I believe this basically Turbo Frames under the hood but just not using the template version like `create.turbo_frame.erb`. Simpler is preferred! This is mainly just how I've built this type of form at work.

Open to hearing suggestions

Reply:
The point is that by using Turbo Drive, you don't need to use frames at all - you just submit the form like you normally would and under the hood it swaps out the body. You can do a normal `render` of ERB content without having to worry about partially updating pages. Really you'd only ever go to turbo frames if performance is critical and/or you don't want to reload the page. And you'd only go to turbo streams if you want a "live update" kind of interface like e-mail or chat.

Reply:
Ahh you are right, I didn't realize we can skip the whole turbo frame approach. I just rendered `:index` again and it worked! I'll work on updated this PR

Reply:
Hey @dorner - I noticed that it does work to some extent, but doesn't work for flash messages when I tried it. The response template includes the partner_users#index content, but not the flash message that exists outside. Been looking at https://www.hotrails.dev/turbo-rails/flash-messages-hotwire and couldn't seem to get it to work without the turbo_stream approach they used in the `.turbo_stream` template.

Is there a way to use just `render :index` to achieve the result we intend? Lemme know what you think